### PR TITLE
Fix Message return types

### PR DESCRIPTION
**Pull request by MGatner**
**Description**
A number of `Message` methods have docblocks for `@return Message` which makes static analysis angry when a child class is returned. This PR switches to `$this` and specifies a `self` return type on the function. I also cleaned up the file a bit but nothing that would affect code.

**Checklist:**
- [X] Securely signed commits
- [X] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- n/a User guide updated
- [X] Conforms to style guide

### CODE REVIEW

**Comment by MGatner** [comment]
Apparently the updated return types resolved this PHPStan error we were ignoring:
> #Call to an undefined method CodeIgniter\\HTTP\\Message::setStatusCode\(\)#

I removed that line and updated the file format to tabs (recommendation from https://ne-on.org) and alphabetized the sequences to be easier to find.

**Comment by MGatner** [comment]
@samsonasik If you would please check over the PHPStan changes for me!

**Comment by samsonasik** [comment]
it seems ok 👍

File: system/HTTP/Message.php [hunk 1]
```diff
@@ -351,6 +328,21 @@ public function prependHeader(string $name, string $value)
...
+	 * Takes a header name in any case, and returns the
+	 * normal-case version of the header.
+	 *
+	 * @param string $name
+	 *
+	 * @return string
+	 */
+	protected function getHeaderName(string $name): string
+	{
+		$lower_name = strtolower($name);
```
Comment (michalsn) [inline system/HTTP/Message.php:341]: I know you're just moving around this method but can we take this opportunity and change the variable name from `$lower_name` to `$lowerName`?

Also, there is one code divider (//------) left after this method.

Comment (MGatner) [reply system/HTTP/Message.php]: Removed the divider and the use of the interim variable.

**Review by michalsn — APPROVED** [review]

**Comment by MGatner** [comment]
I'm fine not finishing the tests again if you are.

File: system/HTTP/Message.php (hunk 1 above)
Comment (samsonasik) [reply system/HTTP/Message.php]: it seems we can use is rector rule for that, that can be to update entire codebase

https://github.com/rectorphp/rector/blob/master/rules/coding-style/src/Rector/Variable/UnderscoreToCamelCaseVariableNameRector.php

**Pull request merged**